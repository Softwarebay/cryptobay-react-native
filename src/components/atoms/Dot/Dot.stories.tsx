import { number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { Dot } from './Dot';

storiesOf('Atoms/Dot', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Default Dot', () => <Dot />)
  .add('Dot With Props', () => (
    <Dot
      size={number('size', 18)}
      color={text('color', '#FEBF32') as `#${string}`}
    />
  ));
