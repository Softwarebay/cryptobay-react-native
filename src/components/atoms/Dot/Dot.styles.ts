import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  dot: {
    borderRadius: 1000,
  },
});
