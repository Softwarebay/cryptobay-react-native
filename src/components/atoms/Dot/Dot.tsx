import React from 'react';
import { View } from 'react-native';

import { styles } from './Dot.styles';

type DotProps = {
  size?: number;
  color?: `#${string}`;
  testID?: string;
};

export const Dot: React.FC<DotProps> = ({ size, color, testID }: DotProps) => {
  const handleStyleByProps = {
    width: size,
    height: size,
    backgroundColor: color,
  };

  return <View testID={testID} style={[styles.dot, handleStyleByProps]} />;
};

Dot.defaultProps = {
  size: 6,
  color: '#FFFFFF',
};
