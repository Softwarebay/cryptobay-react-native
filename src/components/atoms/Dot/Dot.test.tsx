import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Dot } from './Dot';

describe('Dot Atom', () => {
  test('Comonent Snapshot', () => {
    const tree = renderer.create(<Dot />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure Dot component color and size works correctly', () => {
    render(<Dot testID="atom-dot" size={18} color="#FEBF32" />);
    expect(screen.getByTestId('atom-dot')).toHaveStyle({
      width: 18,
      height: 18,
      backgroundColor: '#FEBF32',
    });
  });
});
