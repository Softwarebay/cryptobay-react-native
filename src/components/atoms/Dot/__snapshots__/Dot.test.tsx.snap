// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot Atom Comonent Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 1000,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 6,
        "width": 6,
      },
    ]
  }
/>
`;
